modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -472,12 +474,24 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 		}
 
 		if (is_a_symbol) {
-			if (in_function_name) {
-				in_function_args = true;
+			if (in_function_name && str[j] == '(' && in_function_args == 0) {
+				in_function_args = 1;
+			}
+
+			if (!in_function_name && str[j] == '(' && in_function_args > 0) {
+				in_function_args += 1;
 			}
 
-			if (in_function_args && str[j] == ')') {
-				in_function_args = false;
+			if (in_function_args > 0 && str[j] == ')') {
+				in_function_args -= 1;
+			}
+
+			if (in_function_args > 0) {
+				if (str[j] == '{') {
+					in_function_arg_dicts += 1;
+				} else if (str[j] == '}' && in_function_arg_dicts > 0) {
+					in_function_arg_dicts -= 1;
+				}
 			}
```

Comment:
I haven't tested it, but perhaps this code is a little better?

```c++
			if (in_function_name) {
				if (str[j] == '(' && in_function_args == 0) {
					in_function_args = 1;
				}
			} else if (in_function_args > 0) {
				if (str[j] == '(') {
					in_function_args += 1;
				} else if (str[j] == ')') {
					in_function_args -= 1;
				} else if (str[j] == '{') {
					in_function_arg_dicts += 1;
				} else if (str[j] == '}' && in_function_arg_dicts > 0) {
					in_function_arg_dicts -= 1;
				}
			}
```

Reply:
Yes, this looks more concise. I will take another look today!

Reply:
Commit has been amended with the suggested code improvement. 👍